### PR TITLE
Custom outcome buttons

### PR DESCRIPTION
**Pull request by nickoneill**

### CODE REVIEW

**Comment by nickoneill** [comment]
Expands for different numbers of outcome buttons
screen shot 2017-08-02 at 3 11 16 pm
screen shot 2017-08-02 at 3 12 05 pm

**Comment by nickoneill** [comment]
(this is still WIP, needs adaptation for the new `outcomeModels` field so we can correctly add these outcomes to the impact data)

**Comment by nickoneill** [comment]
Using the new outcome models, tested "upgrading" from old contact log (really it's just supporting older outcome strings as well as new ones), rebased. Ready for review.

File: FiveCalls/FiveCalls/ReportOutcomeOperation.swift [hunk 1]
```diff
@@ -12,23 +12,35 @@ class ReportOutcomeOperation : BaseOperation {
...
+        self.outcome = outcome
     }
     
     override func execute() {
         let config = URLSessionConfiguration.default
         let session = URLSessionProvider.buildSession(configuration: config)
         let url = URL(string: "https://5calls.org/report")!
+
+        // rather than avoiding network calls during debug,
+        // indicate they shouldn't be included in counts
```
Comment (subdigital) [inline FiveCalls/FiveCalls/ReportOutcomeOperation.swift]: 👍

File: FiveCalls/FiveCalls/ContactLog.swift [hunk 2]
```diff
@@ -9,21 +9,21 @@
...
-    case unknown
-}
+//enum ContactOutcome : String {
+//    case contacted
+//    case voicemail = "vm"
+//    case unavailable
+//    
+//    // reserved for cases where we save something on disk that we later don't recognize
+//    case unknown
+//}
```
Comment (subdigital) [inline FiveCalls/FiveCalls/ContactLog.swift]: Time to delete this?

File: FiveCalls/FiveCalls/CallScriptViewController.swift [hunk 3]
```diff
@@ -54,10 +51,9 @@ class CallScriptViewController : UIViewController, IssueShareable {
         self.contactIndex = contactIndex
         title = "Contact \(contactIndex+1) of \(issue.contacts.count)"
 
-        let method = logs.methodOfContact(to: contact.id, forIssue: issue.id)
-        self.resultContactedButton.isSelected = method == .contacted
-        self.resultUnavailableButton.isSelected = method == .unavailable
-        self.resultVoicemailButton.isSelected = method == .voicemail
+        // set the footer height based on how many outcomes there are:
+        // cell height (+ padding) + extra footer spacing
+        footerHeightConstraint.constant = (ceil(CGFloat(issue.outcomes.count) / 2) * OutcomeCollectionCell.cellHeight() + 10) + 40
```
[the remark this answers is not in the document]
Comment (nickoneill) [reply FiveCalls/FiveCalls/CallScriptViewController.swift]: I don't think reaching into the `OutcomeCollectionCell` to get a height is necessarily smelly, but the raw padding and footer spacing added here is def funky. I tried figuring out the content height of the collection view itself but couldn't get it to return something sane. Not sure what the deal was there. Will keep an eye on this for future refactor.

File: FiveCalls/FiveCalls/ContactLog.swift (hunk 2 above)
Comment (nickoneill) [reply FiveCalls/FiveCalls/ContactLog.swift]: Yep, always forget these 🙃

**Pull request merged**